backend/src/common/types/repository.type.ts

```diff
@@ -1,5 +1,8 @@
+import { type UserModel } from '~/bundles/users/user.model.js';
+
 type Repository<T = unknown> = {
     find(): Promise<T>;
+    findByEmail(email: string): Promise<UserModel | null>;
```

Comment:
You change the base Repository type. Therefore every repository class that implements Repository should also have `findByEmail` method. Maybe it's better to add this method to `UserRepository` class as it is without adding UserRepository type to implement from?